### DISCUSSION TITLE
How can I map shift+space to PageUp?

### DISCUSSION THREAD

**Discussion by wgmayer0** [kovidgoyal/kitty#4839] [General]
How can I map shift+space to PageUp?

**Comment by page-down** [comment]
https://sw.kovidgoyal.net/kitty/faq/#how-do-i-map-key-presses-in-kitty-to-different-keys-in-the-terminal-program

```shell
kitty +kitten show_key
```

```conf
map shift+space send_text normal,application \x1b[5~
```

> Is it possible to also use it to scroll in the terminal when not in an application?

You need to write and configure kitten to do this, depending on the conditions to decide whether to scroll up scrollback or send the key escape sequence.

https://sw.kovidgoyal.net/kitty/actions/#action-scroll_page_up

https://sw.kovidgoyal.net/kitty/kittens/custom/#kittens-created-by-kitty-users

https://github.com/yurikhan/kitty-smart-scroll

**Comment by wgmayer0** [comment]
Is it possible to also use it to scroll in the terminal when not in an application?

**Reply by page-down** [reply]
You need to write and configure kitten to do this, depending on the conditions to decide whether to scroll up scrollback or send the key escape sequence.

https://sw.kovidgoyal.net/kitty/actions/#action-scroll_page_up

https://sw.kovidgoyal.net/kitty/kittens/custom/#kittens-created-by-kitty-users

https://github.com/yurikhan/kitty-smart-scroll

**Comment by wgmayer0** [comment]
Thanks, I was able to get it using:

https://github.com/yurikhan/kitty-smart-scroll

With the option: `map shift+space kitten smart_scroll.py scroll_page_up page_up`
